src/main/java/com/microsoft/graph/models/extensions/Multipart.java

```diff
@@ -33,46 +38,132 @@ public Multipart() {
 	 * Get the multipart boundary for use in the request header
 	 * @return the multipart boundary
 	 */
-	public String boundary() {
+	public String getBoundary() {
 		return boundary;
 	}
 	
+	/**
+	 * Set the multipart boundary for use in the request header
+	 * @param boundary The multipart boundary
+	 */
+	public void setBoundary(String boundary) {
+		this.boundary = boundary;
+	}
+	
+	/**
+	 * Get the contentType for use in the request header
+	 * @return the multipart Content-Type
+	 */
+	public String getContentType() {
+		return contentType;
+	}
+	
+	/**
+	 * Set the contentType for use in the request header
+	 * @param contentType The multipart Content-Type
+	 */
+	public void setContentType(String contentType) {
+		this.contentType = contentType;
+	}
+	
 	/**
 	 * Get the Content-Type header to send the multipart request
 	 * @return the multipart header option
 	 */
 	public HeaderOption header() {
-		return new HeaderOption("Content-Type", "multipart/form-data; boundary=\"" + boundary + "\"");
+		return new HeaderOption("Content-Type", contentType + "; boundary=\"" + boundary + "\"");
+	}
+	
+	private void writePartData(String partContent, byte[] byteArray) throws IOException{
+		out.write(partContent.getBytes(MULTIPART_ENCODING));
+		out.write(byteArray);
+		String returnContent = RETURN + RETURN;
+		out.write(returnContent.getBytes(MULTIPART_ENCODING));
+	}
+	
+	private String createPartHeader(Map<String, String> headers, String name, String contentType, String filename) {
```

Comment:
Make this method static if doesn't use parent object state

Reply:
Please add unit test coverage of this method.

Reply:
This method cannot be made static because it uses boundary string which is dependent on parent object state.